### PR TITLE
#207 | create FAQs page

### PR DESCRIPTION
**Pull request by hoavt**
connect #207

### CODE REVIEW

File: docs/faqs.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+Currently, to know the version of teracy-dev which you are using, you need to follow the steps: 
+First, you need to ``git branch`` to know the current branch, then ``git log`` to know the info about the latest commits. Normally, we have the commits about release the new version, for example, "Merge pull request #313 from hoatle/tasks/#312-release-v0.5.0-b3".
+
+We'll provide a simpler way soon at https://github.com/teracyhq/dev/issues/366.
+
+
+After ``vaggrant ssh``, why cannot I run the ``grunt`` or ``npm`` command in the VM?
+------------------------------------------------------------------------------------
+
+The ``grun`` or ``npm`` command is not used in the ``teracy-dev`` latest version anymore. Instead,
```
Comment (hoatle) [inline docs/faqs.rst]: grun => grunt

File: docs/faqs.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+you should install docker and use the ``docker-compose`` command.
```
Comment (hoatle) [inline docs/faqs.rst]: docker => \``docker``

Comment (hoatle) [inline docs/faqs.rst]: by default, `docker` and `docker-compose` is installed by default and you should use these two.

File: docs/faqs.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+3. ``$ ping <the_configured_domain>`` to see if it pings the right VM IP from step 1.
+4. If it's not right, open ``/etc/hosts`` to check and remove wrong entries.
+5. Come back to Step 3 to verify.
+
+
+What should I do when the ``teracy-dev`` version is changed?
+------------------------------------------------------------
+You should run the command ``vagrant destroy``, then ``vagrant up``.
+
+What should I do when the ``vagrant_config_override.json`` file is changed?
```
Comment (hoatle) [inline docs/faqs.rst]: duplicated

File: docs/faqs.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+---------------------------------------------------------------------------
+
+You should  run the command ``$ vagrant reload`` or ``$ vagrant reload --provision``.
+
+What should I do when changing provisioner (Chef, Bash...)?
+-----------------------------------------------------------
+You should run the command `` $ vagrant reload --provision``.
+
+
+What should I do when meeting other  errors, and the problems cannot be solved with ``vagrant reload`` or ``vagrant reload --provision``?
```
Comment (hoatle) [inline docs/faqs.rst]: other errors, 1 white space only

**Comment by hoatle** [comment]
```WARNING: /opt/app/faqs.rst:47: (WARNING/2) Title underline too short.
``/etc/hosts`` is not properly updated when I got errors, after ``$ vagrant destroy`` and ``$ vagrant up``. I end up having multiple same entry that the wrong one is one the top of the file. This leads to wrong DNS to point to the right VM IP. What should I do?
----------------------------------------------------------------------
WARNING: /opt/app/faqs.rst:47: (WARNING/2) Title underline too short.
``/etc/hosts`` is not properly updated when I got errors, after ``$ vagrant destroy`` and ``$ vagrant up``. I end up having multiple same entry that the wrong one is one the top of the file. This leads to wrong DNS to point to the right VM IP. What should I do?
----------------------------------------------------------------------
WARNING: /opt/app/faqs.rst:73: (WARNING/2) Title underline too short.
What should I do when meeting other  errors, and the problems cannot be solved with ``vagrant reload`` or ``vagrant reload --provision``?
-------------------------------------------
WARNING: /opt/app/faqs.rst:73: (WARNING/2) Title underline too short.
What should I do when meeting other  errors, and the problems cannot be solved with ``vagrant reload`` or ``vagrant reload --provision``?
-------------------------------------------``` from https://travis-ci.org/teracyhq/dev/jobs/255610597

@hoavt don't forget to fix these kinds of problems before sending new pull requests.

**Comment by hoavt** [comment]
@hoatle Applied comments. Please check.

File: docs/faqs.rst [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      default: /tmp/vagrant-chef/bbfefdc57119d7552b06b24069242f8a/data_bags => /Users/hoatle/teracy-dev/workspace/teracy-dev/data_bags
+      default: /tmp/vagrant-chef/9b5518c8fee080ca55f1c57179068e17/cookbooks => /Users/hoatle/teracy-dev/workspace/teracy-dev/vendor-cookbooks
+      default: /tmp/vagrant-chef/87b97d785383812081b2ec7e56be857d/cookbooks => /Users/hoatle/teracy-dev/workspace/teracy-dev/main-cookbooks
+      ==> default: Running provisioner: chef_solo...
+      default: Installing Chef (latest)...
+
+Then stop it, ``$ vagrant ssh`` to update this http://askubuntu.com/questions/620317/apt-get-update-stuck-connecting-to-security-ubuntu-com, then ``$ vagrant reload --provision`` again.
+
+
+``/etc/hosts`` is not properly updated when I got errors, after ``$ vagrant destroy`` and ``$ vagrant up``. I end up having multiple same entry that the wrong one is one the top of the file. This leads to wrong DNS to point to the right VM IP. What should I do?
```
Comment (hoatle) [inline docs/faqs.rst]: multiple same entries

the wrong one is on the top of the file

VM's IP address

File: docs/faqs.rst [hunk 6]
```diff
@@ -0,0 +1,68 @@
+FAQs
+====
+This page provides the frequently asked questions (FAQs) about ``teracy-dev``.
+
+Why teracy-dev? Why not just Docker?
+------------------------------------
+Docker is great to work with, it solves a lot of problems on development, distribution, and production deployment. It works great on Linux, however it's very challenging to make it work universal and consistent on Mac and Windows. There are lots of efforts to solve this problem, from Docker themselves and from Docker community, too. But, we haven't achieved that stage yet (until ``teracy-dev``). ``teracy-dev`` is an universal Productive Development Platform With Docker on macOS, Linux and Windows.  With ``teracy-dev``, anyone from any OS platforms could collaborate and help each other with no differences in the development environment, and many other reasons
```
Comment (hoatle) [inline docs/faqs.rst]: But, we haven’t achieved that stage yet (until teracy-dev) => However,

Comment (hoatle) [inline docs/faqs.rst]: an universal => a universal

**Comment by hoavt** [comment]
@hoatle applied your feedback. Please check.

**Pull request merged**